### PR TITLE
Eth2 APIs: Implement ListPool endpoints

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
This PR implements the following endpoints of the API:
```
​/eth​/v1​/beacon​/pool​/attester_slashings
​/eth​/v1​/beacon​/pool​/proposer_slashings
​/eth​/v1​/beacon​/pool​/voluntary_exits
``` 

This PR is part of #7510

### CODE REVIEW

File: beacon-chain/rpc/beaconv1/pool_test.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	slashingsInPool := make([]*ethpb_alpha.AttesterSlashing, 20)
+	v1Slashings := make([]*ethpb.AttesterSlashing, 20)
+	for i := 0; i < len(slashingsInPool); i++ {
+		sl, err := testutil.GenerateAttesterSlashingForValidator(beaconState, privKeys[i], uint64(i))
+		require.NoError(t, err)
+		slashingsInPool[i] = sl
+		marshaledSlashing, err := sl.Marshal()
+		require.NoError(t, err)
+		v1Slashing := &ethpb.AttesterSlashing{}
+		require.NoError(t, proto.Unmarshal(marshaledSlashing, v1Slashing))
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beaconv1/pool_test.go:31]: This conversion can benefit from reusing the code if you put it in the migration package.

File: beacon-chain/rpc/beaconv1/pool_test.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+		},
+		{
+			name:    "All eligible",
+			pending: slashingsInPool,
+			want:    v1Slashings[0:2],
+		},
+		{
+			name:    "Multiple indices",
+			pending: slashingsInPool[3:6],
+			want:    v1Slashings[3:5],
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beaconv1/pool_test.go:52]: Can you elaborate on this test? Why would only some of the slashings be returned? Shouldn't this endpoint return all of the slashings in the pool?

File: beacon-chain/rpc/beaconv1/pool_test.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	}{
+		{
+			name:    "Empty list",
+			pending: []*ethpb_alpha.AttesterSlashing{},
+			want:    []*ethpb.AttesterSlashing{},
+		},
+		{
+			name:    "All eligible",
+			pending: slashingsInPool,
+			want:    v1Slashings[0:2],
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beaconv1/pool_test.go:47]: Shouldn't there be 20 slashings returned? Why only 2?

File: beacon-chain/rpc/beaconv1/pool_test.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+	slashingsInPool := make([]*ethpb_alpha.ProposerSlashing, 5)
+	v1Slashings := make([]*ethpb.ProposerSlashing, 5)
+	for i := 0; i < len(slashingsInPool); i++ {
+		sl, err := testutil.GenerateProposerSlashingForValidator(beaconState, privKeys[i], uint64(i))
+		require.NoError(t, err)
+		slashingsInPool[i] = sl
+		marshaledSlashing, err := sl.Marshal()
+		require.NoError(t, err)
+		v1Slashing := &ethpb.ProposerSlashing{}
+		require.NoError(t, proto.Unmarshal(marshaledSlashing, v1Slashing))
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beaconv1/pool_test.go:84]: Same feedback about reusing code.

File: beacon-chain/rpc/beaconv1/pool_test.go [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	exitsInPool := make([]*ethpb_alpha.SignedVoluntaryExit, 5)
+	v1Exits := make([]*ethpb.SignedVoluntaryExit, 5)
+	for i := 0; i < len(exitsInPool); i++ {
+		exit, err := testutil.GenerateVoluntaryExit(beaconState, privKeys[i], uint64(i))
+		require.NoError(t, err)
+		exitsInPool[i] = exit
+		marshaledExit, err := exit.Marshal()
+		require.NoError(t, err)
+		v1Exit := &ethpb.SignedVoluntaryExit{}
+		require.NoError(t, proto.Unmarshal(marshaledExit, v1Exit))
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beaconv1/pool_test.go:137]: Same feedback about reusing code.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

File: beacon-chain/rpc/beaconv1/pool_test.go (hunk 2 above)
Comment (0xKiwi) [reply beacon-chain/rpc/beaconv1/pool_test.go:52]: Ah, our code for accessing the pool only returns what `MAX_SLASHINGS` per block is set to. Should that be changed?

**Pull request closed without merging**